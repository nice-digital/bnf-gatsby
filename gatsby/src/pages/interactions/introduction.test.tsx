import { render, waitFor, screen } from "@testing-library/react";
import striptags from "striptags";

import InteractionsIntroductionPage, {
	type InteractionsIntroductionPageProps,
} from "./introduction";

const props: InteractionsIntroductionPageProps = {
	data: {
		bnfInteractionsIntroduction: {
			sections: [
				{ content: "<p>I am some content for the introductions page</p>" },
			],
		},
	},
};

describe("InteractionsIntroductionPage", () => {
	it("should set page title", async () => {
		render(<InteractionsIntroductionPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith("Introduction");
		});
	});

	it("should set page meta description", async () => {
		render(<InteractionsIntroductionPage {...props} />);
		await waitFor(() => {
			expect(
				document
					// eslint-disable-next-line testing-library/no-node-access
					.querySelector("meta[name='description']")
					?.getAttribute("content")
			).toEqual(
				"Read about how pharmacodynamic and pharmacokinetic interactions can occur, potential effects of these interactions, and how their severity is graded in the BNF."
			);
		});
	});

	it("should render the supplied content in the page", () => {
		render(<InteractionsIntroductionPage {...props} />);
		expect(
			screen.getByText(
				striptags(props.data.bnfInteractionsIntroduction.sections[0].content)
			)
		).toBeInTheDocument();
	});

	it("should match snapshot for page body", () => {
		render(<InteractionsIntroductionPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});
});
