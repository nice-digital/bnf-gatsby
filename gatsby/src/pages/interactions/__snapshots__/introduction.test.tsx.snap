// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractionsIntroductionPage should match snapshot for page body 1`] = `
<main>
  <div
    class="container"
  >
    <nav
      aria-label="Breadcrumbs"
      role="navigation"
    >
      <ol
        class="breadcrumbs"
      >
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="https://www.nice.org.uk/"
          >
            NICE
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/"
          >
            BNF
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <a
            href="/interactions/"
          >
            Interactions
          </a>
        </li>
        <li
          class="breadcrumbs__crumb"
        >
          <span>
            Introduction
          </span>
        </li>
      </ol>
    </nav>
    <script
      type="application/ld+json"
    >
      {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/interactions/",
        "name": "Interactions"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Introduction"
      }
    }
  ]
}
    </script>
    <div
      class="page-header"
      id="content-start"
    >
      <h1
        class="page-header__heading"
      >
        Interactions: Introduction
      </h1>
    </div>
    <div>
      <p>
        I am some content for the introductions page
      </p>
    </div>
  </div>
</main>
`;
